connectors/anaplan/app/clients.py

```
@@ -0,0 +1,98 @@
+from typing import Any
+
+from application_sdk.clients.base import BaseClient
+from application_sdk.observability.logger_adaptor import get_logger
+
+logger = get_logger(__name__)
+
+
+class AppClient(BaseClient):
+    """Client for App API interactions.
+
+    Handles authentication and HTTP requests to App (Anaplan) APIs.
+    Supports basic authentication with token-based authorization.
+    """
+
+    async def load(self, **kwargs: Any) -> None:
+        """Initialize the client with credentials and necessary attributes.
+
+        Args:
+            **kwargs: Keyword arguments containing credentials and configuration.
+        """
+
+        credentials = kwargs.get("credentials", {})
+        self.credentials = credentials
+        self.host = credentials.get("host", "us1a.app.anaplan.com")
+        self.username = credentials.get("username")
+        self.password = credentials.get("password")
+        self.auth_type = credentials.get("authType", "basic")
+
+        self.auth_token = None
+
+        # Set the auth token and token expiration time
+        await self._get_auth_token()
+
+        # Set default headers for all API calls
+        await self._update_client_headers()
+
+    async def _get_auth_token(self) -> str:
+        """Get authentication token from Anaplan API using basic auth.
+
+        Makes a POST request to Anaplan's authentication endpoint to obtain
+        an authentication token for subsequent API calls.
+
+        Returns:
+            str: The authentication token value.
+
+        Raises:
+            Exception: If authentication fails or token is not found in response.
+
+        Note:
+            Endpoint: POST https://auth.anaplan.com/token/authenticate
+            Expected Response: 201 Created with token information.
+        """
+
+        if self.auth_type == "basic":
+            if not self.username or not self.password:
+                raise Exception(
```

I think you should raise a specific exception like `application_sdk.common.error_codes.ClientError` not `Exception`